File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/C/s081988820_commented.c
#include <stdio.h>

int main() {
    int H, W;
    scanf("%d %d", &H, &W);

    int c[10][10];
    for (int i = 0; i < 10; i++) {
        for (int j = 0; j < 10; j++) {
            scanf("%d", &c[i][j]);
        }
    }

    int min[10];
    for (int i = 0; i < 10; i++) {
        min[i] = c[i][1];
    }

    for (int tc = 0; tc < 10; tc++) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (min[i] > min[j] + c[i][j]) {
                    min[i] = min[j] + c[i][j];
                }
            }
        }
    }

    int ans = 0;
    for (int h = 0; h < H; h++) {
        for (int w = 0; w < W; w++) {
            int A;
            scanf("%d", &A);
            if (A >= 0) {
                ans += min[A];
            }
        }
    }

    printf("%d\n", ans);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s081988820_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)
	var tokens []int
	for scanner.Scan() {
		num, _ := strconv.Atoi(scanner.Text())
		tokens = append(tokens, num)
	}
	H := tokens[0]
	W := tokens[1]
	index := 2

	c := make([][]int, 10)
	for i := range c {
		c[i] = make([]int, 10)
	}
	for i := 0; i < 10; i++ {
		for j := 0; j < 10; j++ {
			c[i][j] = tokens[index]
			index++
		}
	}

	min := make([]int, 10)
	for i := 0; i < 10; i++ {
		min[i] = c[i][1]
	}

	for tc := 0; tc < 10; tc++ {
		for i := 0; i < 10; i++ {
			for j := 0; j < 10; j++ {
				if min[i] > c[i][j]+min[j] {
					min[i] = c[i][j] + min[j]
				}
			}
		}
	}

	ans := 0
	for h := 0; h < H; h++ {
		for w := 0; w < W; w++ {
			A := tokens[index]
			index++
			if A >= 0 {
				ans += min[A]
			}
		}
	}
	fmt.Println(ans)
}

// End of Code
